includes/wikia/parser/templatetypes/handlers/TemplateCleanup.class.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+class TemplateCleanup {
+
+	/**
+	 * This method will remove empty sections headers from article content
+	 *
+	 * @param $parser
+	 * @param $html
+	 *
+	 * @return bool
+	 */
+	public static function doCleanup( $parser, &$html ) {
+		wfProfileIn( __METHOD__ );
+
+		if ( $html && static::shouldBeProcessed() ) {
+			$dom = HtmlHelper::createDOMDocumentFromText( $html );
+
+			$xpath = new DOMXPath( $dom );
+			$headers = [ 'h6', 'h5', 'h4', 'h3', 'h2', 'h1' ];
+			foreach ( $headers as $h ) {
+				$hs = $xpath->query( "//{$h}" );
+				for ( $i = 0; $i < $hs->length; $i++ ) {
+					$current = $hs->item( $i );
+					//skip empty text nodes (spaces and new lines)
```

Comment:
I know it'd give us two return values; Still, I am wondering what would you say about extracting L25-31 along the lines of 
```
list( $nextNode, $emptySiblings ) = $this->skipEmptySiblings($current);
```